test/util/TestHelpers.ts

```diff
@@ -99,11 +99,15 @@ export class ResourceHelper {
       joinFilePath(__dirname, fileLocation),
     );
 
+    const regex = new RegExp(/(.*\/)?([^\s/]+\/?)/u, 'u').exec(slug);
+    const firstSlugPart = regex && regex[1] ? regex[1] : '';
+    const lastSlugPart = regex && regex[2] ? regex[2] : '';
+
     const response: MockResponse<any> = await this.performRequestWithBody(
-      this.baseUrl,
+      new URL(this.baseUrl.toString() + firstSlugPart),
       'POST',
```

Comment:
If you just change this (and the other one below) to be a PUT instead you shouldn't need any of the fancy regexes above. Just doing a PUT to `this.baseUrl + slug` (and potentially some minor changes to make sure slashes are correct) should completely fix the issue. This is the correct way to create resources with a fixed URL anyway since this would also break if we in the future change how we handle slugs again.